src/modal.ts

```diff
@@ -0,0 +1,462 @@
+            if (!active_modals?.length ||
+                active_modals[active_modals.length - 1] == this)
+                this.hide();
+    }
+    clearTimer() {
+        if (this.timer_obj)
+            clearTimeout(this.timer_obj);
+    }
+    containerClicked() {
+        if(this.open && !this.nobgclose)
+            this.hide();
+    }
+    _wrapperClickedEvent(event:any) {
+        event.stopPropagation();
+        if (this.closesel && event.target.matches(this.closesel))
+            this.hide();
+    }
+    show(delay?:number) {
+        this.clearTimer();
+        if (this._open)
+            return;
+        if(delay)
+            this.timer_obj = setTimeout( () => {this.timer_obj=null;
+                                        this.show()}, delay);
+        else {
+            if(this._wrp_init_bounds) {
+               this._wrp_init_bounds = '';
+               this._wrapper_el.style.right = '';
+               this._wrapper_el.style.left = '';
+               this._wrapper_el.style.width = '';
+               this._wrapper_el.style.top = '';
+               this._wrapper_el.style.bottom = '';
+               this._wrapper_el.style.height = '';
+            }
+            this.style.zIndex = DEFAULT_MODAL_ZINDEX;
+            if(this.stack)
+                this.style.zIndex = DEFAULT_MODAL_ZINDEX + active_modals.length;
+            else
+                active_modals.forEach((m:any) => m.hide());
+            if(!this.actualParent)
+                this.actualParent = this.parentElement;
+            this._open = this.open = true;
+            active_modals.push(this);
+            document.body.appendChild(this);
+        }
+    }
+    hide(delay?:number) {
+        this.clearTimer();
+        if (!this._open)
+            return;
+        if (delay)
+            this.timer_obj = setTimeout( () => {this.timer_obj=null;
+                                        this.hide()}, delay);
+        else
+            this.open = false;
+        if (this.animationDur == 0)
+            this.closeEnd();
+
+    }
+    closeEnd() {
+        this._open = this.open;
+        if (!this._open) {
+            this.actualParent?.appendChild(this);
+            let ix = active_modals.indexOf(this);
+            if (ix >= 0)
+                active_modals.splice(ix, 1);
+        }
+    }
+    private _dragStart(event:any) {
+        if (event.target.matches(Object.keys(this._guide_value)))
+            event.preventDefault();
+    }
+    private _touchStart(event:any) {
+        let css_pos = window.getComputedStyle(this._wrapper_el)
+                                 .getPropertyValue("position");
+        if(css_pos == "relative")
+            return;
+        if (!this._wrp_init_bounds)
+            this._wrp_init_bounds = this._wrapper_el.getBoundingClientRect();
+        for (let guide_sel of Object.keys(this._guide_value)) {
+            let el = event.target;
+            while (el) {
+                if (el.matches(guide_sel)) {
```

Comment:
We can use the matches like how you had used it in drag start so that we can avoid the outer loop iterating the guide_value keys.